app/code/Magento/MediaGallery/Plugin/Wysiwyg/Images/Storage.php

```diff
@@ -92,4 +99,27 @@ public function afterDeleteFile(StorageSubject $subject, StorageSubject $result,
 
         return $result;
     }
+
+    /**
+     * Delete media data after the folder delete action from Wysiwyg
+     *
+     * @param StorageSubject $subject
+     * @param mixed $result
+     * @param string $path
+     *
+     * @return null
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterDeleteDirectory(StorageSubject $subject, $result, $path)
+    {
+        if (!is_string($path)) {
+            return $result;
+        }
+
+        $relativePath = $this->filesystem->getDirectoryRead(DirectoryList::MEDIA)->getRelativePath($path);
```

Comment:
`getRelativePath` throws `ValidatorException`. I think it should be logged without interrupting the code execution.

Also, it looks like `getRelativePath` will return the original passed path if an incorrect path was provided, so I'd check if the path exists in media directory not to remove assets if the `StorageSubject` provided an incorrect path for some reason